### PR TITLE
Add tooltip to completions ParameterValue

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
Moved it to a switch so it's easier to read. This will show the tooltips for `ParameterValue`

fixes https://github.com/PowerShell/vscode-powershell/issues/2405

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4801872&bid=15980015)

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs [hunk 1]
```diff
@@ -202,62 +202,65 @@ public void SetCapability(CompletionCapability capability)
             string completionText = completionDetails.CompletionText;
             InsertTextFormat insertTextFormat = InsertTextFormat.PlainText;
 
-            if ((completionDetails.CompletionType == CompletionType.Variable) ||
-                (completionDetails.CompletionType == CompletionType.ParameterName))
+            switch (completionDetails.CompletionType)
```
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: Codacy Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4801872)

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs [hunk 2]
```diff
@@ -202,62 +202,65 @@ public void SetCapability(CompletionCapability capability)
...
+                case CompletionType.Variable:
+                case CompletionType.ParameterValue:
+                case CompletionType.Method:
+                case CompletionType.Property:
+                    detailString = completionDetails.ToolTipText;
+                    break;
+                case CompletionType.ParameterName:
+                    // Look for type encoded in the tooltip for parameters and variables.
+                    // Display PowerShell type names in [] to be consistent with PowerShell syntax
+                    // and now the debugger displays type names.
```
Comment (rkeithhill) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: I think I typo'd that `now`, `how` makes more sense in this comment.

Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]:
```suggestion
                    // and how the debugger displays type names.
```

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: fixed!

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs [hunk 3]
```diff
@@ -202,62 +202,65 @@ public void SetCapability(CompletionCapability capability)
...
-                        !Regex.IsMatch(completionDetails.ToolTipText,
-                            @"^\s*" + escapedToolTipText + @"\s+\["))
+                case CompletionType.Variable:
+                case CompletionType.ParameterValue:
+                case CompletionType.Method:
+                case CompletionType.Property:
+                    detailString = completionDetails.ToolTipText;
+                    break;
+                case CompletionType.ParameterName:
+                    // Look for type encoded in the tooltip for parameters and variables.
```
Comment (rkeithhill) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: Also, now you are just looking at parameter names (not variables).  Should this also case on Variable?

image

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4801872&bid=15980015)

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs (hunk 1 above)
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: Codacy Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4801872)

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs [hunk 4]
```diff
@@ -202,62 +202,65 @@ public void SetCapability(CompletionCapability capability)
...
+                case CompletionType.Folder:
+                    // Insert a final "tab stop" as identified by $0 in the snippet provided for completion.
+                    // For folder paths, we take the path returned by PowerShell e.g. 'C:\Program Files' and insert
+                    // the tab stop marker before the closing quote char e.g. 'C:\Program Files$0'.
+                    // This causes the editing cursor to be placed *before* the final quote after completion,
+                    // which makes subsequent path completions work. See this part of the LSP spec for details:
+                    // https://microsoft.github.io/language-server-protocol/specification#textDocument_completion
+
+                    // Since we want to use a "tab stop" we need to escape a few things for Textmate to render properly.
+                    if(EndsWithQuote(completionText))
```
Comment (rkeithhill) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: Space between `if` and opening `(`.

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs (hunk 3 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: Variables is on line 207?

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: oh I need to move it down gotcha

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs (hunk 4 above)
Comment (rjmholt) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]:
```suggestion
                    if (EndsWithQuote(completionText))
```

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]:
```suggestion
                    if (EndsWithQuote(completionText))
```

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs (hunk 3 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: fixed!

**Comment by TylerLeonhardt** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4801872&bid=15980015)

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs (hunk 1 above)
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/CompletionHandler.cs]: Codacy Issue found: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4801872)

**Pull request merged**